File: app/src/main/java/com/arkapp/partyplanner/ui/finalChecklist/utils/DialogChangeGuest.java
package com.arkapp.partyplanner.ui.finalChecklist.utils;

import android.app.Dialog;
import android.content.Context;
import android.os.Bundle;
import android.text.Editable;
import android.text.TextUtils;
import android.text.TextWatcher;
import android.view.LayoutInflater;

import com.arkapp.partyplanner.data.models.PartyDetails;
import com.arkapp.partyplanner.data.repository.PrefRepository;
import com.arkapp.partyplanner.databinding.DialogChangeGuestBinding;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import static com.arkapp.partyplanner.utils.ViewUtilsKt.setFullWidth;
import static com.arkapp.partyplanner.utils.ViewUtilsKt.setTransparentEdges;

/**
 * Created by Abdul Rehman on 29-05-2020.
 * Contact email - abdulrehman0796@gmail.com
 */

/**
 * This dialog is used for changing the guest count in the checklist screen
 */

public class DialogChangeGuest extends Dialog {

    private final PrefRepository prefRepository;

    public DialogChangeGuest(@NotNull Context context, @NotNull PrefRepository prefRepository) {
        super(context);
        this.prefRepository = prefRepository;
    }

    protected void onCreate(@Nullable Bundle savedInstanceState) {
        final DialogChangeGuestBinding binding = DialogChangeGuestBinding.inflate(LayoutInflater.from(getContext()));
        setContentView(binding.getRoot());

        setFullWidth(getWindow());
        setTransparentEdges(getWindow());

        binding.guestEt.addTextChangedListener(new TextWatcher() {
            @Override
            public void beforeTextChanged(CharSequence s, int start, int count, int after) {
            }

            @Override
            public void onTextChanged(CharSequence s, int start, int before, int count) {
            }

            @Override
            public void afterTextChanged(Editable s) {
                if (!TextUtils.isEmpty(s.toString())) {
                    binding.guest.setError(null);
                    PartyDetails details = prefRepository.getCurrentPartyDetails();
                    details.setPartyGuest(Integer.parseInt(s.toString()));
                    prefRepository.setCurrentPartyDetails(details);
                }
            }
        });


        binding.doneBtn.setOnClickListener(v -> {
            if (binding.guestEt.getText().toString().isEmpty() ||
                    Integer.parseInt(binding.guestEt.getText().toString()) <= 0) {
                binding.guest.setError("Please enter guest count!");
                return;
            }

            dismiss();
        });
    }
}

File: app/src/main/java/com/arkapp/partyplanner/ui/finalChecklist/utils/DialogChangeGuest.kt
package com.arkapp.partyplanner.ui.finalChecklist.utils

import android.app.Dialog
import android.content.Context
import android.os.Bundle
import android.view.LayoutInflater
import androidx.core.widget.doAfterTextChanged
import com.arkapp.partyplanner.data.repository.PrefRepository
import com.arkapp.partyplanner.databinding.DialogChangeGuestBinding
import com.arkapp.partyplanner.utils.setFullWidth
import com.arkapp.partyplanner.utils.setTransparentEdges
import kotlinx.android.synthetic.main.dialog_destination_selection.*

/**
 * Created by Abdul Rehman on 29-05-2020.
 * Contact email - abdulrehman0796@gmail.com
 */
class DialogChangeGuest(context: Context,
                        private val prefRepository: PrefRepository) : Dialog(context) {

    override fun onCreate(savedInstanceState: Bundle?) {
        val binding = DialogChangeGuestBinding.inflate(LayoutInflater.from(context))

        setContentView(binding.root)

        window?.setTransparentEdges()
        window?.setFullWidth()

        binding.guestEt.doAfterTextChanged { text ->
            if (!text.isNullOrEmpty()) {
                binding.guest.error = null
                val details = prefRepository.getCurrentPartyDetails()
                details.partyGuest = text.toString().toInt()
                prefRepository.setCurrentPartyDetails(details)
            }
        }


        doneBtn.setOnClickListener {
            if (binding.guestEt.text.toString().isEmpty() ||
                binding.guestEt.text.toString().toInt() <= 0) {
                binding.guest.error = "Please enter guest count!"
                return@setOnClickListener
            }

            dismiss()
        }
    }
}